crates/common/tedge_utils/src/timers.rs

```diff
@@ -0,0 +1,72 @@
+use std::{collections::HashMap, fmt::Debug, hash::Hash, pin::Pin, time::Duration};
+
+use futures::{stream::FuturesUnordered, Future, StreamExt};
+use tokio::time::{sleep, Instant};
+
+pub type TimerFuture<T> = Pin<Box<dyn Future<Output = T> + Send + Sync>>;
+
+#[derive(Debug, Default)]
+pub struct Timers<K, V> {
+    unfinished_child_op_status_map: HashMap<K, (V, Instant)>,
+    unfinished_child_op_timers: FuturesUnordered<TimerFuture<(K, Instant)>>,
+}
+
+impl<K, V> Timers<K, V>
+where
+    K: Debug + Eq + Hash + Clone + Send + Sync + 'static,
+    V: 'static,
+{
+    pub fn new() -> Self {
+        Timers {
+            unfinished_child_op_status_map: HashMap::new(),
+            unfinished_child_op_timers: FuturesUnordered::new(),
+        }
+    }
+
+    /// So, to "reset" an already scheduled timer, add a new one with the same key which will "logically" replace the former timer.
```

Comment:
This doc comment tells nothing on the main usage of this method and only focuses on what has been tricky to implement ;-) 

```suggestion
    /// Set a timer.
    ///
    /// On call of the `next_timed_out_entry()` method after the given `duration`,  
    /// this timer will remind that the given `value` is expected for the `key`.
    ///
    /// To "reset" an already scheduled timer,
    /// add a new one with the same key which will "logically" replace the former timer.
```